KerbalStuff/blueprints/mods.py

```diff
@@ -36,12 +36,15 @@ def _get_mod_game_info(mod_id):
 
 
 def _restore_game_info():
-    game_id = session.get('gameid') or 'kerbal-space-program'
-    ga = Game.query.filter(Game.id == game_id).order_by(desc(Game.id)).first()
-    if not ga:
-        abort(404)
-    set_game_info(ga)
-    return ga
+    game_id = session.get('gameid')
+    if game_id:
+        game = Game.query.filter(Game.active == True, Game.id == game_id).order_by(desc(Game.id)).first()
```

Comment:
Why is there an `order_by` call here? It looks like we are selecting all the rows with a matching game ID, so all of the game IDs in the returned set will be the same (if it's even possible to have multiple), and then we _sort_ by those values we already know are all the same? Seems like an incomplete thought somewhere in here.
(I realize this conundrum exists in the legacy code, just wondering if we can cure it.)

Reply:
You are completely right. There should only ever be a a single item with that id (I think the database would never allow creating second item with the same id since it's the primary column). So I think we can remove the `order_by()` call completely, and even use `.one()` instead of `.first()`, which throws an exception if there are multiple items.